addons/web/static/src/js/views/basic/basic_model.js

```diff
@@ -1322,10 +1322,14 @@ var BasicModel = AbstractModel.extend({
                         delete record._warning;
                         onchangeDef.resolve(_.keys(changes).concat(Object.keys(result && result.value || {})));
                     }).fail(function () {
-                        self._visitChildren(record, function (elem) {
-                            _.extend(elem, initialData[elem.id]);
-                        });
-                        onchangeDef.resolve({});
+                        if (!options.force_fail) {
```

Comment:
I don't like having a new option, and 2 different behaviors. If the onchange fails, it seems reasonable to always reject the promise. Did you tried this, and did it break tests? It would be interesting to see which use cases need the promise to be resolved anyway...

Reply:
I tested on runbot and just rejecting in all cases don't break any test:

odoo-dev/odoo: https://runbot.odoo.com/runbot/build/3065893
odoo-dev/enterprise: https://runbot.odoo.com/runbot/build/3068957

I updated this PR with this change

I was afraid of doing that because of possibility of changed behavior (after onchange) that don't matter much for kanban view since we reload on onchange failure

Reply:
@aab-odoo @nle-odoo I am not sure to reject in all cases but I did after checking this commit: https://github.com/odoo/odoo/commit/c163351e8a6776e046f04e6c264bf4372499dbba it should not fail, but tests are green after @nle-odoo change it, so I don't it should create issue.

Reply:
@msh-odoo Tests don't cover all scenarios, and in particular, failing use cases are in general badly (or even, not) tested. In master, I definitely go for nle's solution without the options (always reject in the fail/guardedCatch handler), but for a long time stable version, I'd rather do the safe patch which consist in changing the behavior only in the case that we want to fix. Please add a comment indicating that this is a local patch for stable version (maybe with the opw id or something).

Reply:
@aab-odoo force pushed my version of fix, please check it